module/metrics/network.go

```diff
@@ -121,6 +123,20 @@ func NewNetworkCollector() *NetworkCollector {
 			Name:      "inbound_connection_count",
 			Help:      "the number of inbound connections of this node",
 		}),
+
```

Comment:
Do you know why the connections in libp2p have the concept of direction? Shouldn't the connections be bidirectional since they communicate back and forth to each other?

> One unstaked node can potentially have more than one connection with the upstream Staked AN (one for the pubsub and then one for the direct one-to-one) but Libp2p actually will reuse the connection if one already exist.

To be clear, you're saying that while there may be multiple `Connection` objects, there is only one underlying TCP connection. Is that correct?

Still not sure I understand why the connections have directions: https://github.com/onflow/flow-go/blob/2273886f496c49ede3b7c0d350a2aab5b6a3b862/network/p2p/connManager.go#L89

Reply:
> actually we need this as a metric - len(libP2PNode.pubSub.ListPeers(UnstakedSyncCommittee). This will give us the number of unstaked nodes talking to this staked node right now.

This will be a useful metric for sure, but I'm not sure this is sufficient. ListPeers will only tell us our direct peers on that channel. but the staked AN may be connected to multiple other unstaked nodes as metadata peers.

I would argue actually that this metric is less useful than the connection count, because by default the configuration for gossipsub limits our direct peers for each topic to between Dlow and Dhigh, so `len(ListPeers())` will always be within this range.

Reply:
Connections have direction to indicate who initiated the connection. Once the connection is established it is used for bi-directional traffic.

Reply:
Okay, makes sense. Thabks!

So, do u still want me to add the metric for `len(libP2PNode.pubSub.ListPeers(UnstakedSyncCommittee)`?

If you think it will be useful then I can easily add it. I'm just not sure whether it will actually provide useful information due to the reasons above.

Reply:
lets not add it now. I am not sure though if we can correctly answer the question: "how many unstaked nodes are being served by this staked access node but the connection counts?" we can start with what you have though